enterprise/internal/campaigns/resolvers/changeset_event_connection_test.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/sourcegraph/sourcegraph/internal/db/dbtesting"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/github"
+)
+
+func TestChangesetEventConnectionResolver(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	ctx := backend.WithAuthzBypass(context.Background())
+	dbtesting.SetupGlobalTestDB(t)
+
+	userID := insertTestUser(t, dbconn.Global, "changeset-event-connection-resolver", true)
+
+	now := time.Now().UTC().Truncate(time.Microsecond)
+	clock := func() time.Time {
+		return now.UTC().Truncate(time.Microsecond)
+	}
+	store := ee.NewStoreWithClock(dbconn.Global, clock)
+	rstore := repos.NewDBStore(dbconn.Global, sql.TxOptions{})
+
+	repo := newGitHubTestRepo("github.com/sourcegraph/sourcegraph", 1)
+	if err := rstore.UpsertRepos(ctx, repo); err != nil {
+		t.Fatal(err)
+	}
+
+	campaign := &campaigns.Campaign{
+		Name:            "my-unique-name",
+		NamespaceUserID: userID,
+		AuthorID:        userID,
+	}
+	if err := store.CreateCampaign(ctx, campaign); err != nil {
+		t.Fatal(err)
+	}
+
+	changeset := createChangeset(t, ctx, store, testChangesetOpts{
+		repo: repo.ID,
+		// We don't need a spec because we don't query for fields that would
+		// require it
+		currentSpec:         0,
+		externalServiceType: "github",
+		publicationState:    campaigns.ChangesetPublicationStateUnpublished,
+		externalReviewState: campaigns.ChangesetReviewStatePending,
+		ownedByCampaign:     campaign.ID,
+		campaign:            campaign.ID,
+		metadata: &github.PullRequest{
+			TimelineItems: []github.TimelineItem{
+				{Type: "PullRequestCommit", Item: &github.PullRequestCommit{
+					Commit: github.Commit{
+						OID: "d34db33f",
+					},
+				}},
+				{Type: "LabeledEvent", Item: &github.LabelEvent{
+					Label: github.Label{
+						ID:    "label-event",
+						Name:  "cool-label",
+						Color: "blue",
+					},
+				}},
+			},
+		},
+	})
+
+	addChangeset(t, ctx, store, campaign, changeset.ID)
+
+	s, err := graphqlbackend.NewSchema(&Resolver{store: store}, nil, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	changesetAPIID := string(marshalChangesetID(changeset.ID))
+	nodes := []apitest.ChangesetEvent{
+		{
+			ID:        string(marshalChangesetEventID(1)),
+			Changeset: struct{ ID string }{ID: changesetAPIID},
+			CreatedAt: marshalDateTime(t, now),
+		},
+		{
+			ID:        string(marshalChangesetEventID(2)),
+			Changeset: struct{ ID string }{ID: changesetAPIID},
+			CreatedAt: marshalDateTime(t, now),
+		},
+	}
+
+	tests := []struct {
+		firstParam      int
+		wantHasNextPage bool
+		wantTotalCount  int
+		wantNodes       []apitest.ChangesetEvent
+	}{
+		{firstParam: 1, wantHasNextPage: true, wantTotalCount: 2, wantNodes: nodes[:1]},
+		{firstParam: 2, wantHasNextPage: false, wantTotalCount: 2, wantNodes: nodes},
+		{firstParam: 3, wantHasNextPage: false, wantTotalCount: 2, wantNodes: nodes},
+	}
+
+	for _, tc := range tests {
+		t.Run(fmt.Sprintf("First %d", tc.firstParam), func(t *testing.T) {
```

Comment:
Yep, it's me, Mr. Nitpick, and I'm here to suggest some things you can freely drag'n'drop into `/dev/null`:

```suggestion
		t.Run(fmt.Sprintf("first=%d", tc.firstParam), func(t *testing.T) {
```